src/common/page-queries/account.ts

```diff
@@ -6,18 +6,47 @@ import { DEFAULT_LIST_LIMIT } from '@common/constants';
 import { MempoolTransactionListResponse } from '@stacks/stacks-blockchain-api-types';
 import { InfoQueryKeys } from '@store/info';
 
-export function getPrincipalFromCtx(ctx: NextPageContext) {
+export async function getPrincipalFromCtx(ctx: NextPageContext) {
   const { query } = ctx;
-  const principal: string = query?.principal as string;
+  const { bnsApi } = await getApiClients(ctx);
+  let principal: string = query?.principal as string;
+
+  // TODO: Is this the correct place for this?
+  const { namespaces } = await bnsApi.getAllNamespaces();
+  if (namespaces.filter(namespace => principal.endsWith(namespace)).length) {
+    bnsApi.getAllNamespaces();
+
+    try {
+      const res = await bnsApi.getNameInfo({
+        name: principal,
+      });
+
+      principal = res.address;
+    } catch {
+      principal = '';
+    }
+  }
+
```

Comment:
Perhaps here we could check for valid STX addresses, and return early when it matches, avoiding unnecessary calls

Reply:
@kyranjamie So I had it implemented previously as a simple `.endsWith` check, happy to switch back to that method.

But this approach allows us to resolve `.btc` addresses etc

Reply:
Yes, I saw this. As I understand, this fn is pulling the query info from the request, and it _could_ contain a valid STX address.

If this is the case, it's a wasted effort to request all this bns info, right? Couldn't we just return early? We could use the `c32check` lib to validate the input. There's already a helper for this, actually, `validateStacksAddress`